packages/astro/src/core/errors/errors.ts

```diff
@@ -74,7 +74,7 @@ export class AstroError extends Error {
 	}
 
 	static is(err: unknown): err is AstroError {
-		return (err as AstroError).type === 'AstroError';
+		return err != null && (err as AstroError).type === 'AstroError';
```

Comment:
More a stylistic thing, but I guess we might as well use optional chaining for all of these?
```suggestion
		return (err as AstroError)?.type === 'AstroError';
```